packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,217 +7,372 @@ excerpt: >-
+       <head>
+           <title>Spring Boot and SAML</title>
+           <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+       </head>
+       <body>
+           <h1>Welcome</h1>
+           <p>You are successfully logged in as <span sec:authentication="name"></span></p>
+           <p>Your email address is <span th:text="${emailAddress}"></span>.</p>
+           <p>Your authorities are <span sec:authentication="authorities"></span>.</p>
+           <h2>All Your Attributes</h2>
+           <dl th:each="userAttribute : ${userAttributes}">
+               <dt th:text="${userAttribute.key}"></dt>
+               <dd th:text="${userAttribute.value}"></dd>
+           </dl>
 
-Use the following command to start Tomcat:
+           <form th:action="@{/logout}" method="post">
+               <button id="logout" type="submit">Logout</button>
+           </form>
+       </body>
+   </html>
+   ```
 
-```bash
-/Library/Tomcat/bin/startup.sh
-```
+1. Create a `src/main/resources/application.yml` file to contain your metadata URI.
 
-**7. Start the Application**
+   ```yaml
+   spring:
+     security:
+       saml2:
+         relyingparty:
+           registration:
+             okta:
+               assertingparty:
+                 metadata-uri: <your-metadata-uri>
+   ```
 
-Load the Spring SAML application by opening this link: `http://localhost:8080/spring-security-saml2-sample/saml/discovery?entityID=http%3A%2F%2Flocalhost%3A8080%2Fspring-security-saml2-sample%2Fsaml%2Fmetadata&returnIDParam=idp`
+### Run the app and authenticate
 
-> **Note:** Links in the app aren't functional yet because we haven't configured any IDPs. Full app functionality is completed after the **Configure Spring Security SAML to work with Okta** section.
+1. Run your Spring Boot app. You can do this from your IDE, or as follows using the command line:
 
-Here's what it should look like: ![App Running](/img/spring-security-saml-intro.png "App Running")
+   ```shell
+   ./gradlew bootRun
+   ```
 
-## Configure Okta to work with Spring Security SAML
+1. Open `http://localhost:8080` in your favorite browser and sign in with the credentials you used to create your account. You should see a successful result in your browser.
+
+   <div class="three-quarter border">
+
+   ![Login success](/img/saml/spring-security-saml-login.png)
+
+   </div>
+
+If you try to sign out, it won't work. You'll fix that in the next section.
+
+### Add a sign out feature
+
+Spring Security's SAML support has a [sign out feature](https://docs.spring.io/spring-security/reference/servlet/saml2/logout.html) that requires configuration.
+
+1. Edit your application on Okta and navigate to **General** > **SAML Settings** > **Edit**.
+
+1. Continue to the **Configure SAML** step and **Show Advanced Settings**. Before you can enable single logout, you'll have to create and upload a certificate to sign the outgoing logout request.
```

Comment:
I think developers are more familiar with logout as a concept. I think it makes sense to keep it as logout, especially since our Admin Console calls it "Enable Single Logout" and not "Enable Single Sign-Out".

Reply:
@mraible It is messy and I know, but here is the text from our style guide. Basically, the UI hasn't caught up to the correct style yet but you can use log in or log lout to match something in the UI  even you have to say something like "Set the Enable LogOut URI field to reflect where the user will be redirected to after they sign out".

> - Verb: sign in, sign out (no hyphen). Sign in with a username and password.
> - Adjective: sign-in, sign-out (hyphen). In Identity Engine, there is no security image on the Sign-In Widget.
> - Noun: don’t use this term as a noun. Instead use the adjective form and a descriptive noun. Sign-in attempts from malicious IP addresses are displayed in the System Log.
> 
> Don’t use sign on or log in interchangeably with sign in. 
> 
> Acceptable uses of sign-on: sign-on policy (app or Okta), Single Sign-On (SSO)
> Acceptable uses of login: API endpoints (such as Identity Providers or social logins)
> 
> If the UI element uses a term that we specifically advise against, use that version only for task topic instructions, when you have to reference the field or button name. Follow our best practices for concept topics and release notes.